onnxscript/tests/function_libs/torch_lib/ops_test.py

```diff
@@ -102,52 +102,49 @@ def _get_rtol_atol_by_dtype(dtype: torch.dtype) -> tuple[Any, Any]:
 
 class TestFunctionValidity(unittest.TestCase):
     def test_all_script_functions_are_onnx_functions(self):
-        functions = set()
-        for func_with_wrangler in ops_test_data.OPINFO_FUNCTION_MAPPING_SCRIPTED.values():
-            func, _ = _split_function_and_wrangler(func_with_wrangler)
-            functions.add(func)
-
-        # TODO(justinchuby): Add from the registry
-        for func in functions:
-            if not isinstance(func, onnxscript.OnnxFunction):
-                raise AssertionError(
-                    f"'{func}' is not an OnnxFunction. Was it decorated with '@torch_op'? "
-                    "If the function is trace_only, please specify trace_only=True "
-                    "in the TorchLibOpInfo entry."
-                )
+        for info in ops_test_data.TESTED_TORCHLIB_OPS:
+            if info.trace_only:
+                continue
+            with self.subTest(name=info.op_info_name):
+                func = info.op
+                if not isinstance(func, onnxscript.OnnxFunction):
+                    raise AssertionError(
+                        f"'{func}' is not an OnnxFunction. Was it decorated with '@torch_op'? "
+                        "If the function is trace_only, please specify trace_only=True "
+                        "in the TorchLibOpInfo entry."
+                    )
 
     def test_all_trace_only_functions_are_not_onnx_functions(self):
-        for func_with_wrangler in ops_test_data.OPINFO_FUNCTION_MAPPING_TRACE_ONLY.values():
-            func, _ = _split_function_and_wrangler(func_with_wrangler)
-            if isinstance(func, onnxscript.OnnxFunction):
-                raise AssertionError(
-                    f"'{func.name}' is an OnnxFunction. "
-                    "If the function is not trace_only, please remove trace_only=True "
-                    "in the TorchLibOpInfo entry."
-                )
+        for info in ops_test_data.TESTED_TORCHLIB_OPS:
+            if not info.trace_only:
+                continue
+            with self.subTest(name=info.op_info_name):
+                func = info.op
+                if not isinstance(func, onnxscript.TracedOnnxFunction):
```

Comment:
If we don't have to explicit trace_only attribute, we don't need this second guard to make sure it's TracedOnnxFunction.

Reply:
I wanted to make it harder to use trace_only (because we eventually want to script all functions, as far as we know now) so that authors know they are making a conscious choice.